pysatNASA/instruments/icon_mighti.py

```diff
@@ -132,68 +132,77 @@ def clean(self):
-        wind_vars = ['Zonal_Wind', 'Zonal_Wind_Error',
-                     'Meridional_Wind', 'Meridional_Wind_Error']
-        ver_vars = ['Fringe_Amplitude', 'Fringe_Amplitude_Error',
-                    'Relative_VER', 'Relative_VER_Error']
-        if wind_flag not in self.variables:
-            wind_flag = '_'.join(('ICON_L22', wind_flag))
-            ver_flag = '_'.join(('ICON_L22', ver_flag))
-            wind_vars = ['ICON_L22_' + var for var in wind_vars]
-            ver_vars = ['ICON_L22_' + var for var in ver_vars]
-        min_val = {'clean': 1.0,
-                   'dusty': 0.5}
-        if self.clean_level in ['clean', 'dusty']:
+            _clean_vars(wind_vars, wind_flag, min_val[self.clean_level])
+            _clean_vars(ver_vars, ver_flag, min_val[self.clean_level])
+
+        elif self.tag.find('vector') >= 0:
+            # vector winds area
+            wind_flag = 'Wind_Quality'
+            ver_flag = 'VER_Quality'
+            wind_vars = ['Zonal_Wind', 'Zonal_Wind_Error',
+                         'Meridional_Wind', 'Meridional_Wind_Error']
+            ver_vars = ['Fringe_Amplitude', 'Fringe_Amplitude_Error',
+                        'Relative_VER', 'Relative_VER_Error']
+            if wind_flag not in self.variables:
+                wind_flag = '_'.join(('ICON_L22', wind_flag))
+                ver_flag = '_'.join(('ICON_L22', ver_flag))
+                wind_vars = ['ICON_L22_' + var for var in wind_vars]
+                ver_vars = ['ICON_L22_' + var for var in ver_vars]
+            min_val = {'clean': 1.0,
+                       'dusty': 0.5}
             # find location with any of the flags set
-            for var in wind_vars:
-                self[var] = self[var].where(self[wind_flag]
-                                            >= min_val[self.clean_level])
-            for var in ver_vars:
-                self[var] = self[var].where(self[ver_flag]
-                                            >= min_val[self.clean_level])
-
-    elif self.tag.find('temp') >= 0:
-        # neutral temperatures
-        var = 'Temperature'
-        saa_flag = 'Quality_Flag_South_Atlantic_Anomaly'
-        cal_flag = 'Quality_Flag_Bad_Calibration'
-        if saa_flag not in self.variables:
-            id_str = self.inst_id.upper()
-            saa_flag = '_'.join(('ICON_L1_MIGHTI', id_str, saa_flag))
-            cal_flag = '_'.join(('ICON_L1_MIGHTI', id_str, cal_flag))
-            var = '_'.join(('ICON_L23_MIGHTI', id_str, var))
-        if self.clean_level in ['clean', 'dusty']:
+            _clean_vars(wind_vars, wind_flag, min_val[self.clean_level])
+            _clean_vars(ver_vars, ver_flag, min_val[self.clean_level])
+
+        elif self.tag.find('temp') >= 0:
+            # neutral temperatures
+            var = 'Temperature'
+            saa_flag = 'Quality_Flag_South_Atlantic_Anomaly'
+            cal_flag = 'Quality_Flag_Bad_Calibration'
+            if saa_flag not in self.variables:
```

Comment:
Please excuse my icon ignorance, but can pysat load different data levels (L1, L2 etc) for icon? When is it expected to find that the flags need 'ICON_L1_MIGHTI' prepended to them?

Reply:
We don't have Level 1 data yet, but some of the data products copy over variables from the Level 1 files, which is why some variables have the prefix in the variable name